### PR TITLE
fix(log): Allow passing optional params for logs

### LINKED ISSUE #8657: Add an option to print full log.
**Is your feature request related to a problem? Please describe.**
Sometimes I want to track an issue by using `console.log`,  the console could only print part of the logs, if its too long it would be omitted like this.
image

**Describe the solution you'd like**
Add an option to print full log

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by jdnichollsc**
## PR Checklist

- [x] The PR title follows our guidelines: https://github.com/NativeScript/NativeScript/blob/master/tools/notes/CONTRIBUTING.md#commit-messages.
- [ ] There is an issue for the bug/feature this PR is for. To avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it.
- [x] You have signed the [CLA](http://www.nativescript.org/cla).
- [x] All existing tests are passing: https://github.com/NativeScript/NativeScript/blob/master/tools/notes/DevelopmentWorkflow.md#running-unit-tests-application.
- [ ] Tests for the changes are included - https://github.com/NativeScript/NativeScript/blob/master/tools/notes/WritingUnitTests.md.

## What is the current behavior?
Important logs don't accept other arguments.

## What is the new behavior?
Allow passing arguments for important logs

Fixes/Implements/Closes #8657.

Thanks for your collaboration! <3

### CODE REVIEW

**Comment by farfromrefug** [comment]
@jdnichollsc awesome PR!

**Review by NathanWalker — APPROVED** [review]

File: packages/core/profiling/index.ts [hunk 1]
```diff
@@ -6,11 +6,11 @@ export function uptime() {
 	return global.android ? (<any>org).nativescript.Process.getUpTime() : (<any>global).__tns_uptime();
 }
 
-export function log(message: string): void {
+export function log(message: string, ...optionalParams: any[]): void {
 	if ((<any>global).__nslog) {
-		(<any>global).__nslog('CONSOLE LOG: ' + message);
+		(<any>global).__nslog('CONSOLE LOG: ' + message, ...optionalParams);
```
Comment (NathanWalker) [inline packages/core/profiling/index.ts:11]: Given this passes through NSLog, not sure if we need to define an override (in the ios runtime) as mentioned here:
https://stackoverflow.com/a/42674570

@rigor789 may know?

The console.log below definitely makes sense and shouldn't be an issue but the __nslog, I'm not sure.

Comment (NathanWalker) [reply packages/core/profiling/index.ts:11]: Looks like here it's a single string arg: https://github.com/NativeScript/ns-v8ios-runtime/blob/master/AppWithModules/Frameworks/TNSWidgets.xcframework/ios-arm64/TNSWidgets.framework/Headers/TNSProcess.h#L27

@jdnichollsc did you try this running against latest @nativescript/ios 8.0.0? just curious if you found the optionalParams additions make it through the global.__nslog okay?

Comment (jdnichollsc) [reply packages/core/profiling/index.ts:11]: You're right! NSLog only works with one argument 🤦

Comment (NathanWalker) [reply packages/core/profiling/index.ts]: Excellent thanks for updating!

Comment (jdnichollsc) [reply packages/core/profiling/index.ts]: You're welcome!

**Pull request merged**